src/unstract/sdk/embedding.py

```diff
@@ -23,13 +24,13 @@ def __init__(self, tool: BaseTool, tool_settings: dict[str, str] = {}):
 
     def get_embedding(
         self, adapter_instance_id: Optional[str] = None
-    ) -> Optional[BaseEmbedding]:
+    ) -> BaseEmbedding:
         adapter_instance_id = (
             adapter_instance_id
             if adapter_instance_id
             else self.embedding_adapter_instance_id
         )
-        if adapter_instance_id is not None:
+        if adapter_instance_id:
```

Comment:
@gaya3-zipstack do you remember why we need this check? If its not needed (please add a TODO if you're not going to take this up immediately)
- The snippet above this line can be removed altogether
```
adapter_instance_id = (
            adapter_instance_id
            if adapter_instance_id
            else self.embedding_adapter_instance_id
        )
```
- The `adapter_instance_id` type can be `str`
- The `else` block associated can be removed as well

I can't seem to find usages for `.get_embedding()`
![image](https://github.com/Zipstack/unstract-sdk/assets/117059509/8861faf0-2471-4850-8595-8cfd7e0bcfe6)
Here every usage passes an `adapter_instance_id`
![image](https://github.com/Zipstack/unstract-sdk/assets/117059509/f55b7623-8d04-46de-83be-e7f4d0f7d6c9)

Reply:
This was added by @muhammad-ali-e to support some tool's way of using the API. I had mentioned about this to him. Not sure if anything will get affected. Need hos confirmation.

Reply:
I think we directly passing the adapter_instance_id now. That time some tools doesn't have that settings. We can remove this on next sdk release.